src/hotspot/share/utilities/tribool.hpp

```diff
@@ -40,11 +40,14 @@ class TriBool{
 
  public:
   TriBool() : _value(0) {}
-  TriBool(bool value) : _value(((u1)value) | 2) {}
+  TriBool(bool value) : _value(value) {
+    _value = _value | 2;
```

Comment:
You can also do like line 39 and use `(u1)(expr) & 3` or `((expr & 3u)` also seems to work.

Reply:
Or cast the `2` to `(u1)2` ?

Reply:
It does seem strange that this code is using a bit-field, but still using & | and shift.  Why not represent the high and low bit with two separate bit-fields of width 1?

Reply:
I tried to change the class to be int value : 1, init: 1;  the changes got a bit ugly when fixing the TriBoolArray functions.

Reply:
expr & 3 is the wrong expression.  This wants to set the 'not-default' bit, not and in the 'not-default' bit if it's set in the value.  Thankfully, tribool has a gtest.
```
tribool.hpp:43:43: warning: conversion from 'unsigned int' to 'unsigned char:2' may change value [-Wconversion]
   43 |   TriBool(bool value) : _value((u1)value  | (u1)2) {
      |                                ~~~~~~~~~~~^~~~~~~
```
Casting 2 to u1 doesn't work because | still promotes to unsigned int.